joint_teapot/utils/joj3.py

```diff
@@ -0,0 +1,104 @@
+import csv
+import json
+import os
+from datetime import datetime
+from typing import Any, Dict
+
+from joint_teapot.utils.logger import logger
+
+
+def generate_scoreboard(score_file_path: str, scoreboard_file_path: str) -> None:
+    if not score_file_path.endswith(".json"):
+        logger.error(
+            f"Score file should be a .json file, but now it is {score_file_path}"
+        )
+        return
+    if not scoreboard_file_path.endswith(".csv"):
+        logger.error(
+            f"Scoreboard file should be a .csv file, but now it is {scoreboard_file_path}"
+        )
+        return
+
+    # Load the csv file if it already exists
+    if os.path.exists(scoreboard_file_path):
+        with open(scoreboard_file_path, newline="") as file:
+            reader = csv.reader(file)
+            rows = list(reader)
+        columns = rows[0]
+        data = rows[1:]
+    else:
+        columns = [
+            "",
+            "last_edit",  # This is just to make changes in the file so that it can be pushed.
+            # Only used in development stage. Will be removed in the future.
+            "total",
+        ]
+        data = []
+
+    column_updated = [False] * len(columns)  # Record wether a score has been updated
+    # Update data
+    with open(score_file_path) as json_file:
+        scoreboard: Dict[str, Any] = json.load(json_file)
+
+    student = f"{scoreboard['studentname']} {scoreboard['studentid']}"
```

Comment:
What's the format of the `scoreboard`? Is there an example?

Reply:
```json
{
  "studentname": "TestingStudent",
  "studentid": "520370000000",
  "stagerecords": [
    {
      "stagename": "clangtidy/sillycode",
      "stageresults": [
        {
          "name": "clangtidy",
          "results": [
            {
              "score": -185,
              "comment": "### Test results summary\n\n1. codequality-unchecked-malloc-result: 0\n2. codequality-no-global-variables: 0\n3. codequality-no-header-guard: 0\n4. codequality-no-fflush-stdin: 0\n5. readability-function-size: 0\n6. readability-duplicate-include: 0\n7. readability-identifier-naming: 0\n8. readability-redundant: 0\n9. readability-misleading-indentation: 0\n10. readability-misplaced-array-index: 0\n11. cppcoreguidelines-init-variables: 0\n12. bugprone-suspicious-string-compare: 0\n13. google-global-names-in-headers: 0\n14. clang-diagnostic: 1\n15. clang-analyzer: 0\n16. misc: 9\n17. performance: 5\n18. others: 185\n"
            },
            ...
          ]
        },
        ...
      ]
    },
    ...
}
```

Reply:
I think that's a mistake and should be better called as `scorefile`